### PR TITLE
[FEATURE] Add usage statistics instrumentation to Legacy UserConfigurableProfiler execution

### PR DESCRIPTION
**Pull request by alexsherstinsky**
### Scope
* Update events `JSON` schema to include the payload for running `UserConfigurableProfiler.build_suite()`.
* Add event payload construction and event emitting method to `UserConfigurableProfiler` (the V2 Legacy Profiler).
* Update unit tests for `UserConfigurableProfiler` to include assertions on the emitted usage statistics events.
* Add unit tests for `JSON` schema and new event messages for the instrumented `UserConfigurableProfiler` component.

Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
- JIRA: ANT-20/ANT-24
-
-

After submitting your PR, CI checks will run and @ge-cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/en/latest/contributing/testing.html#contributing-testing-writing-unit-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/profile/user_configurable_profiler.py [hunk 1]
```diff
@@ -211,9 +213,56 @@ def build_suite(self) -> ExpectationSuite:
             self.profile_dataset._expectation_suite = ExpectationSuite(suite_name)
 
         if self.semantic_types_dict:
-            return self._build_expectation_suite_from_semantic_types_dict()
+            expectation_suite = self._build_expectation_suite_from_semantic_types_dict()
+        else:
+            expectation_suite = self._profile_and_build_expectation_suite()
```
Comment (anthonyburdi) [inline great_expectations/profile/user_configurable_profiler.py:218]: ❤️

File: great_expectations/profile/user_configurable_profiler.py [hunk 2]
```diff
@@ -211,9 +213,56 @@ def build_suite(self) -> ExpectationSuite:
...
+    def _send_usage_stats_message(self):
+        profile_dataset_type: str = str(type(self.profile_dataset))
+        if "Dataset" in profile_dataset_type:
+            profile_dataset_type = "Dataset"
+        elif "Batch" in profile_dataset_type:
+            profile_dataset_type = "Batch"
+        elif "Validator" in profile_dataset_type:
+            profile_dataset_type = "Validator"
+        else:
+            raise ValueError(
```
Comment (anthonyburdi) [inline great_expectations/profile/user_configurable_profiler.py:233]: Maybe you could add this error message to the assert statement in the `__init__()` method?

File: tests/integration/usage_statistics/test_usage_statistics_messages.py [hunk 3]
```diff
@@ -2005,17 +2183,17 @@ def test_generate_messages_with_defaults():
             {
                 "event": "cli.checkpoint.run",
                 "event_payload": {},
-                "ge_version": "0.11.9.manual_testing",
+                "ge_version": "0.13.18.manual_testing",
```
Comment (anthonyburdi) [inline tests/integration/usage_statistics/test_usage_statistics_messages.py:2186]: ❤️

**Review by anthonyburdi — APPROVED** [review]
LGTM! One non blocking suggestion.

File: great_expectations/profile/user_configurable_profiler.py (hunk 2 above)
Comment (alexsherstinsky) [reply great_expectations/profile/user_configurable_profiler.py:233]: @anthonyburdi Can be done, I just wanted to only minimally alter @talagluck 's implementation.

**Pull request merged**